### PR TITLE
Fix TGAT Attention Bug

### LINKED ISSUE #52: Device Transfer on Deduplication without Nbrs

### PR DESCRIPTION
**Pull request by Jacob-Chmura**
# Purpose
When running multi-hop sampling, there will often be times when the nbr mask is (partially) full. The temporal multi-head attention module will take this mask into account when computing the attention tensor: https://github.com/tgm-team/tgm/blob/5b4949e5cb06b5fe94e8f292b9412c908d7f0795/tgm/nn/attention.py#L64-L76

But the key value heads propagate the features as is: https://github.com/tgm-team/tgm/blob/5b4949e5cb06b5fe94e8f292b9412c908d7f0795/tgm/nn/attention.py#L54-L57

Since nbr tensors are initialized with `torch.empty()` these garbage values mess with network gradients and lead to `nans`.

## Solution
Safely initialize all features with `torch.zeros`.

## Relevant Prs
Close #52

### CODE REVIEW

**Comment by shenyangHuang** [comment]
> # Purpose
> When running multi-hop sampling, there will often be times when the nbr mask is (partially) full. The temporal multi-head attention module will take this mask into account when computing the attention tensor:
> 
> https://github.com/tgm-team/tgm/blob/5b4949e5cb06b5fe94e8f292b9412c908d7f0795/tgm/nn/attention.py#L64-L76
> 
> But the key value heads propagate the features as is:
> 
> https://github.com/tgm-team/tgm/blob/5b4949e5cb06b5fe94e8f292b9412c908d7f0795/tgm/nn/attention.py#L54-L57
> 
> Since nbr tensors are initialized with `torch.empty()` these garbage values mess with network gradients and lead to `nans`.
> 
> ## Solution
> Safely initialize all features with `torch.zeros`.
> 
> ## Relevant Prs
> Close #52

good catch, is this from the TGB (PyG) implementation? if so, we can submit this as PR for the PyG code to fix it too

**Review by shenyangHuang — APPROVED** [review]
